mkosi/config.py

```diff
@@ -680,6 +680,10 @@ def from_namespace(cls, ns: argparse.Namespace) -> "MkosiArgs":
             if k in inspect.signature(cls).parameters
         })
 
+    def merge(self, other: "MkosiArgs") -> "MkosiArgs":
+        """Return a new MkosiArgs with attributes from other overwriting the ones from this instance."""
+        return dataclasses.replace(self, **dataclasses.asdict(other))
```

Comment:
How does this work? Won't this just replace the entire dataclass since `MkosiArgs` and `MkosiConfig` must always have all fields?

Reply:
It can't in this way. That was a brain fart. Factored out the JSON to dict conversion and used it here, this way a partial object represented as a dict can be loaded and overwrite `self`, which I would usually assume to be empty.